### PR TITLE
fix highwatermark and logendoffset error

### LINKED ISSUE #329: [BUG] HighWaterMark and LogEndOffset error
**Describe the bug**
For kafka, HighWaterMark and LogEndOffset should be the max offset + 1

### PR DESCRIPTION
**Pull request by aloyszhang**
fix #329 
fix highwatermark and logendoffset error

### CODE REVIEW

**Comment by BewareMyPower** [comment]
Could you add some tests? IMO this PR may fix #276

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java [hunk 1]
```diff
@@ -107,6 +107,14 @@ public static final long offsetAfterBatchIndex(long offset) {
         return offset;
     }
 
+    public static long getHighWaterMark(ManagedLedger managedLedger) {
```
Comment (jiazhai) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java:110]: Hi @aloyszhang Seems these 2 methods are the same, we provided a wrapper on getCurrentOffset, but the inside code not changed, Would you please help provide more info on how could this fix the issue?

**Comment by BewareMyPower** [comment]
Please rebase to latest master to resolve the conflicts and these two lines need to fix: https://github.com/streamnative/kop/blob/a59d745d1602373c4279fab73530ee072c92df9e/tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaApisTest.java#L320-L321 by just removing the comment and modifying the expected value in `assertEqual`.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java (hunk 1 above)
Comment (aloyszhang) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java:110]: `getCurrentOffset`  gives the max offset for the current time, and `getHighWaterMark` and `getLogEndOffset` will give  `getCurrentOffset` + 1.

**Comment by aloyszhang** [comment]
@BewareMyPower  I have rebase the master branch, PTAL.   
And for #276 , I found there is something wrong with the logic of `ListOffset` when doing binary search.
I'll open an issue later and test for `ListOffset` will be added with the new issue.

**Comment by BewareMyPower** [comment]
@aloyszhang As I said before, `testConsumerListOffsetLatest` needs to fix.

**Review by BewareMyPower — APPROVED** [review]

**Comment by aloyszhang** [comment]
@BewareMyPower  I noticed your comment about the ListOffset test , but the test is not cover all condition.  You can see #334

**Pull request merged**